dice-where-downloader-lib/src/main/java/technology/dice/dicewhere/downloader/destination/local/LocalFileAcceptor.java

```diff
@@ -39,6 +40,10 @@ public StreamConsumer<Void> getStreamConsumer(
         LOG.debug("Destination directory already exists");
       }
       Files.copy(stream, destination, StandardCopyOption.REPLACE_EXISTING);
+      if ((!noMd5Check) && (!originalFileMd5.matches(stream.md5()))) {
+        LOG.error("MD5 mismatch. Deleting destination file");
+        Files.delete(destination);
```

Comment:
In theory this should never be invoked I think because we always try to download "latest" from s3 to local which should never be corrupted, correct? I was just wondering if this can have some repercussions for new instances starting up as I saw that we use DownloadExecutionResult.fileLocation in id after we download and due to this line the file would not be available under that location.